riocli/apply/parse.py

```diff
@@ -379,57 +380,57 @@ def _register_object(self, data):
             return
 
     def _load_file_content(self, file_name, is_value=False, is_secret=False):
-        """Load the file content and return the parsed data.
+        """Load and optionally render a template file, then parse as JSON/YAML."""
+        path = Path(file_name)
+        is_template = path.suffix == ".j2"
+
+        # Resolve real extension (e.g., config.yaml.j2 → .yaml)
+        if is_template:
+            ext = Path(path.stem).suffix.lower()  # second-to-last extension
+        else:
+            ext = path.suffix.lower()
 
-        When the file is a template, render it using values or secrets.
-        """
         try:
             if is_secret:
                 data = run_bash(f"sops -d {file_name}")
             else:
-                with open(file_name) as f:
+                with open(file_name, "r") as f:
                     data = f.read()
         except Exception as e:
-            raise Exception(f"Error loading file {file_name}: {str(e)}")
-
-        # When the file is a template, render it using
-        # values or secrets.
-        if not (is_value or is_secret):
-            if self.environment or file_name.endswith(".j2"):
-                try:
-                    template = self.environment.from_string(data)
-                except Exception as e:
-                    raise Exception(f"Error loading template {file_name}: {str(e)}")
-
-                template_args = self.values
-
-                if self.secrets:
-                    template_args["secrets"] = self.secrets
-
-                try:
-                    data = template.render(**template_args)
-                except Exception as ex:
-                    raise Exception(f"Failed to parse {file_name}: {str(ex)}")
+            raise Exception(f"Error loading file {file_name}: {e}")
 
-                file_name = file_name.rstrip(".j2")
+        # Handle Jinja2 templating if needed
+        if not is_secret and self.environment:
```

Comment:
suggestion: Consider removing the self.environment check because it is now unconditionally initialized in the `__init__()`.

Reply:
We can support templating for the secret files as well I think. There is no harm.